lib/cylc/scheduler.py

```diff
@@ -1123,9 +1123,8 @@ def configure_pyro(self):
     def load_suiterc(self, reconfigure):
         """Load and log the suite definition."""
 
-        old_config = False
         if reconfigure:
-            old_config = self.config
+            old_tasks = set(self.config.get_task_name_list())
```

Comment:
This is better, but just out of curiosity - does it actually make a measurable difference to the memory high water mark (or otherwise?)?  I thought about this on my first look, but figured the reference to the full `old_config` would only live for the rest of the function call, and garbage collection isn't going to happen instantly anyway.